gapic-common/lib/gapic/rest/threaded_enumerator.rb

```diff
@@ -38,16 +37,26 @@ def initialize &block
 
         Thread.new do
           @block.call @in_q, @out_q
-
-          @in_q.close
-          @out_q.close
+        rescue StandardError => e
+          @out_q.push e
         end
       end
 
       def next
         @in_q.enq :next
         chunk = @out_q.deq
-        raise StopIteration if chunk.nil?
+
+        if chunk.is_a? StandardError
+          @out_q.close
+          @in_q.close
```

Comment:
How about also clearing the out_q ? It can be misleading receiving an error about dangling content and having old stuff in the queue.